Lib/asyncio/base_events.py

```diff
@@ -1285,9 +1285,8 @@ async def create_datagram_endpoint(self, protocol_factory,
                 addr_infos = {}  # Using order preserving dict
                 for idx, addr in ((0, local_addr), (1, remote_addr)):
                     if addr is not None:
-                        assert isinstance(addr, tuple) and len(addr) == 2, (
-                            '2-tuple is expected')
-
+                        if not (isinstance(addr, tuple) and len(addr) == 2):
+                            raise ValueError(f'addr should be a tuple of length 2 not {addr!r}')
```

Comment:
There is no addr parameter. There are local_addr and remote_addr. Depending on family they can be None, str, etc, not only 2-tuples.

I think that TypeError is more appropriate here.